### PR TITLE
Update discovery.md on mDNS Information

### PR DESCRIPTION
**Pull request by axkar**
Add information about:
  - mDNS alias
  - https://network.local

## Checklist

Tick *relevant* boxes, this PR is-a or has-a:

- [ ] Bugfix
  - [ ] Regression tests
  - [ ] ChangeLog updates (for next release)
- [ ] Feature
  - [ ] YANG model change => revision updated?
  - [ ] Regression tests added?
  - [ ] ChangeLog updates (for next release)
  - [ ] Documentation added?
- [ ] Test changes
  - [ ] Checked in changed Readme.adoc (make test-spec)
  - [ ] Added new test to group Readme.adoc and yaml file
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (please detail in commit messages)
- [ ] Build related changes
- [x] Documentation content changes
  - [x] ChangeLog updated (for major changes)
- [ ] Other (please describe):

### CODE REVIEW

File: doc/discovery.md [hunk 1]
```diff
@@ -171,6 +171,110 @@ linux-pc:# avahi-browse -ar
...
+
+With the help of the mdns-alias, mDNS functionality allows the device
+to be accessible as *infix.local* rather than its full hostname 
+(e.g., infix-00-00-00.local). This alias works seamlessly on a network
+with a single Infix device. However, if multiple Infix devices are
+present (e.g., infix-00-00-00.local, infix-01-01-01.local), each device
+requires a unique hostname to avoid conflicts, as infix.local would not
+uniquely identify multiple devices. The shortened names can also be
+used for actions like pinging or establishing an SSH connection:
+
```
Comment (troglobit) [inline doc/discovery.md:229]: This needs to be simplified.   The `mdns-alias` program is a background helper process we shouldn't mention to the user, if Avahi adds native support for this in the future we will replace `mdns-alias` with that.

Also, the `infix.local` is not shortform of `infix-xx-yy-zz.local`, it's hard-coded to the `DEFAULT_HOSTNAME` from the file `/etc/os-release` + `.local`.  The file `/etc/os-release` in turn is unique for each build, for a generic Infix build `DEFAULT_HOSTNAME=infix`, or `DEFAULT_HOSTNAME=ix` for the minimal builds.  Customer builds will have other defaults.

We need to explain that and the fact that only one device can "own" the alias.  For mDNS it's the first device on the LAN that retains the privilege.  All other devices joining in on the LAN will realize they are not allowed to advertise `infix.local`.

**Review by troglobit — CHANGES_REQUESTED** [review]
A few comments I'd like you to address before we merge this.  

First, I'd really like to see the netbrowse section at the very beginning of this document.  The document will come to life quicker with the nice picture of of the browser showing a listing of multiple devices.

Additionally, I'd like you to overhaul the whole page to not use the default Qemu MAC address (00-00-00) and instead use something else, e.g., c0-ff-ee or similar that we use elsewhere.  Check with J-O for good examples.

Also, a representative example for netbrowse would be to have more than one device in the listing, at least one more, and use a different hostname than the default -- because that's one of the first things an administrator changes.  Again, check with J-O for good name examples, we use descriptive names in the tests so should we here in the docs.

File: doc/discovery.md [hunk 2]
```diff
@@ -171,6 +171,110 @@ linux-pc:# avahi-browse -ar
...
+Additionally, *avahi-resolve-host-name* can be used to verify domain
+name mappings for IP addresses. By default, it translates from IPv4
+addresses. This function allows users to confirm that addresses are
+mapped correctly.
+
+```
+linux-pc:# avahi-resolve-host-name infix-00-00-00.local
+infix-00-00-00.local	10.0.1.1
+linux-pc:#
```
[the remark this answers is not in the document]
Comment (jovatn) [reply doc/discovery.md:183]: Perhaps just my linux pc, but "host" command is unable to resolve network.local, this despite that the browser is able to do it ... :-/
```
linux-pc:~ $ host network.local
Host network.local not found: 2(SERVFAIL)
linuxpc:~ $ 

```

[the remark this answers is not in the document]
Comment (jovatn) [reply doc/discovery.md:183]: Need some help to understand this. I tried this on target. nslookup fails, but ssh succeeds ...

```
admin@infix-bar2:~$ nslookup infix-foo1.local
;; connection timed out; no servers could be reached

admin@infix-bar2:~$ ssh infix-foo1.local
The authenticity of host 'infix-foo1.local (192.168.122.73)' can't be established.
ED25519 key fingerprint is SHA256:RPDPWfsC1Rm+LXqCL77fSBpHnavvGFsAzIjWqqZMfJw.
This key is not known by any other names.
Are you sure you want to continue connecting (yes/no/[fingerprint])? yes
Failed to add the host to the list of known hosts (/home/admin/.ssh/known_hosts).
(admin@infix-foo1.local) Password: 
.-------.
|  . .  | Infix -- a Network Operating System
|-. v .-| https://kernelkit.org
'-'---'-'

Run the command 'cli' for interactive OAM

admin@infix-foo1:~$ 
```

[the remark this answers is not in the document]
Comment (jovatn) [reply doc/discovery.md:183]: https://askubuntu.com/questions/627906/why-is-my-etc-hosts-file-not-queried-when-nslookup-tries-to-resolve-an-address

So I guess nslookup does not use the /etc/nsswitch.conf

File: doc/discovery.md (hunk 1 above)
Comment (jovatn) [reply doc/discovery.md]: Fixing parts of this. Hope it is good enough for now, but comments are of course welcome!

**Comment by jovatn** [comment]
Plenty of commits. :-/
Anyhow, I hope it is now in shape for a new review.

**Review by jovatn — APPROVED** [review]
LGTM. Additional documentation on netbrowse can be added later.

**Pull request merged**